### PR TITLE
Add some comments for Server re-org sample

### PR DESCRIPTION
**Pull request by maria-robobug**

### CODE REVIEW

**Review by osfameron — COMMENTED** [review]
Briefly discussed with Maria - not sure about these comments:
Looks like they are replicating a lot of the information that could be factored in the Asciidoc?

I'm also not sure who the comment is for? Is this the right abstraction level for the typical developer? If we comment anything, would it be the *shape* of e.g. `result` or `row`? An example JSON output or similar.

File: modules/hello-world/examples/n1ql-hello-world.js [hunk 1]
```diff
@@ -7,7 +7,11 @@ async function go() {
   })
 
   // tag::n1ql-query[]
+  // To execute the query, we call the query() function on our cluster object and store the result.
```
Comment (simon-dew) [inline modules/hello-world/examples/n1ql-hello-world.js]:
```suggestion
  // Call the query() function on the cluster object and store the result.
```

File: modules/hello-world/examples/n1ql-hello-world.js [hunk 2]
```diff
@@ -7,7 +7,11 @@ async function go() {
...
   const result = await cluster.query('SELECT "Hello World" as greeting')
+
+  // The result object will include an array of rows found.
+  // We iterate over the rows to access our result data and print it to the terminal.
```
Comment (simon-dew) [inline modules/hello-world/examples/n1ql-hello-world.js]:
```suggestion
  // Iterate over the rows to access result data and print to the terminal.
```

**Review by simon-dew — COMMENTED** [review]
Thanks @maria-robobug , this is great. I've just cut them down slightly to save space. As @osfameron says if further explanation is needed it can go in the AsciiDoc.

**Comment by simon-dew** [comment]
> I'm also not sure who the comment is for?

I would say, a developer who has a good knowledge of the language, and maybe some prior knowledge of traditional SQL, but is new to Couchbase and needs to know what are the methods, objects, or functions offered by the SDK.

**Comment by maria-robobug** [comment]
> As @osfameron says if further explanation is needed it can go in the AsciiDoc.

Sounds good, just have some minor questions:
- When we say this should go in the AsciiDoc are we referring to pages in the **docs-server** repo? 
- Or will we reference pages/partials from the **docs-sdk-{insert-language-here}** repos like we are doing with these code snippets?

**Comment by simon-dew** [comment]
> When we say this should go in the AsciiDoc

I mean that these code snippets might be included anywhere. The including page should include the appropriate level of explanation. So in the docs server pages there might be more explanation; in the SDK pages there might be less.

**Comment by simon-dew** [comment]
I'm updating [couchbase/docs-server#2249](https://github.com/couchbase/docs-server/pull/2249) to include a one- or two-line explanation of these code snippets that is (hopefully) appropriate for the developer how-to guide. I'll add you as a reviewer when it's ready @maria-robobug

**Pull request merged**